jwst/outlier_detection/outlier_detection_ifu.py

```diff
@@ -53,203 +49,258 @@ def __init__(self, input_models, reffiles=None, **pars):
+            nanindx = np.where(nanminarr)
+
+        # Update DQ flag
+        for i in range(len(self.input_models)):
+            model = self.input_models[i]
+            sci = model.data
+            dq = model.dq
+
+            detector = model.meta.instrument.detector.lower()
+            # only use data from the same detector
+            if detector == uq_det[idet]:
+
+                # There could be a large number of pixels with a sci value of NaN
+                # but the dq flag of DO_NOT_USE has not been set.
+                # This can occur in Non-science regions of the detector.
+                check = np.where(
+                    np.logical_and(~np.bitwise_and(dq, dqflags.pixel['DO_NOT_USE']).astype(bool),
+                                   np.isnan(sci)))
+                log.debug("Number of pixels DQ was not set to DO_NOT_USE and Sci array was Nan{} ".
+                          format(len(check[0])))
+                # set all pixels with dq = DO_NOT_USE to have sci values of Nan
+                bad = np.where(np.bitwise_and(dq, dqflags.pixel['DO_NOT_USE']).astype(bool))
+                sci[bad] = np.nan
+
+                # Basic setting outliers: flagging those at are found in from Percentage cut
+                sci[indx] = np.nan
+                dq[indx] = np.bitwise_or(dq[indx], dqflags.pixel['DO_NOT_USE'])
+                dq[indx] = np.bitwise_or(dq[indx], dqflags.pixel['OUTLIER'])
+
+                nadditional = 0
+                # Second level of setting outliers: flagging pixels were minarr was a Nan
+                # This will also catch pixels that have a sci of Nan but the DQ flags did
+                # not have DO_NOT_USE set
+                if ifu_second_check:
+                    # For counting purposes, count the number of science values that were valid (not Nan)
+                    # after basic flagging in the nanminarr region that will now  be flagged as a Nan.
+                    additional = np.where(~np.isnan(sci[nanindx]))
+                    nadditional = len(additional[0])
+
+                    sci[nanindx] = np.nan
+                    dq[nanindx] = np.bitwise_or(dq[nanindx], dqflags.pixel['DO_NOT_USE'])
+                    dq[nanindx] = np.bitwise_or(dq[nanindx], dqflags.pixel['OUTLIER'])
+                    log.info("Number of outlier pixels flagged main ifu outlier flagging: {} on detector {} ".format(
+                        len(indx[0]), uq_det[idet]))
+                    log.info("Number of outlier pixels flagged in second check: {} on detector {} ".format(
+                        nadditional, uq_det[idet]))
+
+                total_bad = len(indx[0]) + nadditional
+                percent_cr = total_bad / (model.data.shape[0] * model.data.shape[1]) * 100
+                log.info(f"Total #  pixels flagged as outliers: {total_bad} ({percent_cr:.2f}%)")
+                # update model
+                model.dq = dq
+                model.data = sci
```

Comment:
There's no need for this assignment since the `sci` and `dq` arrays were not copies but references so `model.data` and `model.dq` were modified along the way.